content/en/platform/corda/4.9/enterprise/release-notes-enterprise.md

```diff
@@ -17,6 +17,82 @@ weight: 10
 
 # Corda Enterprise Edition 4.9 release notes
 
+## Corda Enterprise Edition 4.9.7 release notes
+
+Corda Enterprise Edition 4.9.7 is a patch release of Corda Enterprise focused on resolving issues.
+
+### Upgrade recommendation
+
+As a developer or node operator, you should upgrade to the [latest released version of Corda](../enterprise.html) as soon as possible. The latest Corda Enterprise release notes are on this page, and for the latest upgrade guide, refer to [Upgrading a CorDapp or node](upgrading-index.md).
+
+### Improvements
+
+* A new or restarted peer node coming online and connecting to a node for the first time can significantly slow message processing from other peers on the node to which it connects.  With this improvement, new peers coming online get a dedicated thread on the node they connect to and do not delay message processing for existing peer-to-peer connections on the receiving node.
+
+* Debug logging of the Artemis server has been added.
+
+* The thread dump on heartbeat feature was causing excess log output. This has now been improved: we now only log thread stack traces that appear to be stuck and no longer repeatedly log them.
+
+* Previously, when configured to use confidential identities and the Securosys PrimusX HSM, it was possible for Corda to fail to generate a wrapped key-pair for a new confidential identity. This would cause a temporary key-pair to be leaked, consuming resource in the HSM. This issue occurred when:
+
+  * the Securosys HSM was configured in a master-clone cluster
+  * the master HSM had failed and Corda had failed-over to use the clone HSM
+  * there was an attempt to create a transaction using confidential identities
+
+  The issue is now resolved. When generating a wrapped key-pair the temporary key-pair is not persisted in the HSM and thus cannot be leaked.
+
+  On applying this update the PrimusX JCE should be upgraded to version 2.3.4 or later.
```

Comment:
@chriscochrane Is the above mandatory. As per what you mentioned earlier it is not mandatory, i.e. the upgrade.

Reply:
No it is not mandatory, but for 'ideal behaviour' the JCE should be upgraded.

Without an upgrade - ephemeral keys are stored in the HSM but then automatically garbage-collected within 24h.
With an upgrade - no ephemeral keys are stored in the HSM at all.